plugin/fees/fees.go

```diff
@@ -199,13 +283,108 @@ func (fp *FeePlugin) collectFeesByPolicy(policyIdString string) error {
+				if !fee.Collected {
+					checkAmount += fee.Amount
+				}
+			}
+			if checkAmount != feeHistory.FeesPendingCollection {
+				fp.Log(logrus.ErrorLevel, "Fees pending collection amount does not match the sum of the fees")
+				return fmt.Errorf("fees pending collection amount does not match the sum of the fees")
+			}
+
+			fp.executeFeeCollection(ctx, publicKey, feeHistory.Fees)
+
+		} else {
+			fp.Log(logrus.InfoLevel, "No fees pending collection for public key: ", publicKey)
+		}
+	}
+
 	return nil
 }
 
-// Collects fee data by ... Should only be called by HandleFeeCollections
-func (fp *FeePlugin) collectFeesByAll() error {
-	// TODO: implement logic
+func (fp *FeePlugin) executeFeeCollection(ctx context.Context, ecdsaPublicKey string, feeIds []verifierapi.FeeDto) error {
+
+	fp.Log(logrus.DebugLevel, "Executing fee collection for public key: ", ecdsaPublicKey)
+
+	//Get fee policy document
+	feePolicies, err := fp.db.GetAllPluginPolicies(ctx, ecdsaPublicKey, vtypes.PluginVultisigFees_feee, true)
+	fp.Log(logrus.DebugLevel, "Fee policies: ", len(feePolicies))
+	if err != nil {
+		fp.Log(logrus.DebugLevel, "Failed to get fee policies: ", err)
+		return fmt.Errorf("failed to get fee policies: %w", err)
+	}
+	if len(feePolicies) == 0 {
+		fp.Log(logrus.DebugLevel, "No fee policies found")
+		return fmt.Errorf("no fee policies found")
+	} else if len(feePolicies) > 1 {
+		fp.Log(logrus.DebugLevel, "Multiple fee policies found")
+		return fmt.Errorf("multiple fee policies found")
+	}
+	feePolicy := feePolicies[0]
+	fp.Log(logrus.DebugLevel, "Fee policy: ", feePolicy)
+
+	//Here we check if the fee collection is already in progress for the public key.
+	feeRun, err := fp.db.CreateFeeRun(ctx, feePolicy.ID, types.FeeRunStateDraft, feeIds)
+	if err != nil {
+		fp.Log(logrus.DebugLevel, "Failed to create fee run: ", err)
+		return fmt.Errorf("failed to create fee run: %w", err)
+	}
+	fp.Log(logrus.DebugLevel, "Fee run created: ", feeRun)
+
+	//Get vault and check it exists
+
+	vaultFileName := vcommon.GetVaultBackupFilename(ecdsaPublicKey, vtypes.PluginVultisigFees_feee.String())
+	vaultContent, err := fp.vaultStorage.GetVault(vaultFileName)
+	if err != nil {
+		// TODO some real error handling here
+		fp.Log(logrus.DebugLevel, "Failed to get vault: ", err)
+		return fmt.Errorf("failed to get vault: %w", err)
+	}
+	if vaultContent == nil {
+		// TODO some real error handling here
+		fp.Log(logrus.DebugLevel, "Vault not found")
+		return fmt.Errorf("vault not found")
+	}
+
+	keySignRequests, err := fp.ProposeTransactions(feePolicy)
+	if err != nil {
+		fp.Log(logrus.DebugLevel, "Failed to propose transactions: ", err)
+	}
+	fp.Log(logrus.DebugLevel, "keySignRequest: ", keySignRequests)
+
+	var eg errgroup.Group
+	for _, keySignRequest := range keySignRequests {
+		req := keySignRequest
+		eg.Go(func() error {
+			return fp.initSign(ctx, req, feePolicy)
+		})
+	}
+	err = eg.Wait()
+	if err != nil {
+		fp.logger.WithError(err).Error("eg.Wait")
```

Comment:
in the same file , sometimes you use `fp.logger`, sometims you use `fp.Log`

Consistency....